src/connectors/okta.py

```diff
@@ -0,0 +1,93 @@
+from runners.helpers import db, log
+from runners.config import CONNECTOR_METADATA_TABLE
+
+import datetime
+import requests
+
+CONNECTION_OPTIONS = {
+    'subdomain': {type: str},
+    'api_key': {type: str},
+}
+
+OKTA_LANDING_TABLE = """
+(raw VARIANT,
+ event_time TIMESTAMP_LTZ
+)
+"""
+
+
+def connect(name, options):
+    comment = f"""
+---
+name: {name}
+source: okta
+api_key_secret: {options['api_key']}
+subdomain: {options['subdomain']}
+"""
+
+    results = {}
+    try:
+        db.create_table(name=name+"_CONNECTION", cols=OKTA_LANDING_TABLE, comment=comment)
+        results['events_table'] = 'success'
+    except Exception as e:
+        results['events_table'] = 'failure'
+        results['exception'] = e
+        return results
+
+    return results
+
+
+def populate(name, options):
+    headers = {'Accept': 'application/json', 'Content-Type': 'application/json', 'Authorization': f'SSWS {options["api_key"]}'}
+    url = f"https://{options['subdomain']}.okta.com/api/v1/logs"
+
+    timestamp_query = f"""
+    SELECT EVENT_TIME from DATA.{name}_CONNECTION
+    WHERE EVENT_TIME IS NOT NULL
+    order by EVENT_TIME desc
+    limit 1
+    """
+
+    metadata = {'START_TIME': datetime.datetime.utcnow(), 'TYPE': 'Okta', 'QUERY_NAME': f'{name}_Okta'}
+
+    try:
+        _, ts = db.connect_and_fetchall(timestamp_query)
```

Comment:
since you're discarding the ctx, simple fetch would do well here and something like

    ts = next(db.fetch(timestamp_query), [None])[0]
    if ts:
        ...

should work nicely